src/ejabberd_socket.erl

```diff
@@ -83,26 +83,27 @@ start(Module, SockMod, Socket, Opts) ->
             SocketData = #socket_state{sockmod = SockMod,
                                        socket = Socket,
                                        receiver = RecRef},
-            case Module:start({?MODULE, SocketData}, Opts) of
-                {ok, Pid} ->
-                    case SockMod:controlling_process(Socket, Receiver) of
-                        ok ->
-                            ok;
+            case SockMod:controlling_process(Socket, Receiver) of
```

Comment:
`start` function is way to long and nested.

Reply:
this change is required to enable just_tls ssl backend for legacy TLS connection. (we must ensure that receiver process is the owner of the port before we start tls)

Reply:
So, this should be in a comment before `controlling_process` call.